extension/src/experiments/model/index.test.ts

```diff
@@ -426,33 +477,49 @@ describe('ExperimentsModel', () => {
     expect((experimentsModel as any).useFiltersForSelection).toBe(false)
   })
 
-  it('should fetch branch params', () => {
-    const model = new ExperimentsModel('', buildMockMemento())
-    model.transformAndSet(outputFixture, false)
+  it('should fetch branch params', async () => {
+    const model = new ExperimentsModel(
+      '',
+      buildMockMemento(),
+      mockedInternalCommands
+    )
+    await model.transformAndSet(outputFixture, false)
 
     const branchParams = model.getExperimentParams('main')
     expect(definedAndNonEmpty(branchParams)).toBe(true)
   })
 
-  it('should fetch workspace params', () => {
-    const model = new ExperimentsModel('', buildMockMemento())
-    model.transformAndSet(outputFixture, false)
+  it('should fetch workspace params', async () => {
+    const model = new ExperimentsModel(
+      '',
+      buildMockMemento(),
+      mockedInternalCommands
+    )
+    await model.transformAndSet(outputFixture, false)
 
     const workspaceParams = model.getExperimentParams(EXPERIMENT_WORKSPACE_ID)
     expect(definedAndNonEmpty(workspaceParams)).toBe(true)
   })
 
-  it("should fetch an experiment's params", () => {
-    const model = new ExperimentsModel('', buildMockMemento())
-    model.transformAndSet(outputFixture, false)
+  it("should fetch an experiment's params", async () => {
+    const model = new ExperimentsModel(
+      '',
+      buildMockMemento(),
+      mockedInternalCommands
+    )
+    await model.transformAndSet(outputFixture, false)
 
     const experimentParams = model.getExperimentParams('exp-e7a67')
     expect(definedAndNonEmpty(experimentParams)).toBe(true)
   })
 
-  it("should fetch an empty array if the experiment's params cannot be found", () => {
-    const model = new ExperimentsModel('', buildMockMemento())
-    model.transformAndSet(outputFixture, false)
+  it("should fetch an empty array if the experiment's params cannot be found", async () => {
+    const model = new ExperimentsModel(
+      '',
+      buildMockMemento(),
+      mockedInternalCommands
+    )
+    await model.transformAndSet(outputFixture, false)
```

Comment:
Adding a promise to the model's `transformAndSet` caused me to add more `await`s to multiple tests.

Reply:
another reason to get the data outside.